textdb/textdb-textql/src/test/java/edu/uci/ics/textdb/textql/statements/predicates/KeywordExtractPredicateTest.java

```diff
@@ -0,0 +1,162 @@
+     * @param matchingFields The matchingFields value of the KeywordExtractPredicate.
+     * @param keywords The keywords value of the KeywordExtractPredicate.
+     * @param matchingType The matchingType value of the KeywordExtractPredicate.
+     */
+    private void assertConstructorGettersSetters(List<String> matchingFields, String keywords, String matchingType){
+        KeywordExtractPredicate keywordExtractPredicate;
+        
+        // Check the constructor with the arguments for initialization
+        keywordExtractPredicate = new KeywordExtractPredicate(matchingFields, keywords, matchingType);
+        Assert.assertEquals(keywordExtractPredicate.getMatchingFields(), matchingFields);
+        Assert.assertEquals(keywordExtractPredicate.getKeywords(), keywords);
+        Assert.assertEquals(keywordExtractPredicate.getMatchingType(), matchingType);
+        
+        // Check the constructor with no arguments
+        keywordExtractPredicate = new KeywordExtractPredicate();
+        Assert.assertEquals(keywordExtractPredicate.getMatchingFields(), null);
+        Assert.assertEquals(keywordExtractPredicate.getKeywords(), null);
+        Assert.assertEquals(keywordExtractPredicate.getMatchingType(), null);
+        
+        // Check all the setters with the given values
+        keywordExtractPredicate.setMatchingFields(matchingFields);
+        Assert.assertEquals(keywordExtractPredicate.getMatchingFields(), matchingFields);
+        keywordExtractPredicate.setKeywords(keywords);
+        Assert.assertEquals(keywordExtractPredicate.getKeywords(), keywords);
+        keywordExtractPredicate.setMatchingType(matchingType);
+        Assert.assertEquals(keywordExtractPredicate.getMatchingType(), matchingType);
+        
+        // Check all the setters with null values
+        keywordExtractPredicate.setMatchingFields(null);
+        Assert.assertEquals(keywordExtractPredicate.getMatchingFields(), null);
+        keywordExtractPredicate.setKeywords(null);
+        Assert.assertEquals(keywordExtractPredicate.getKeywords(), null);
+        keywordExtractPredicate.setMatchingType(null);
+        Assert.assertEquals(keywordExtractPredicate.getMatchingType(), null);
+    }
+
+    /**
+     * Test the getOperatorBean method.
+     * Build a KeywordExtractPredicate, invoke the getOperatorBean and
+     * check whether a KeywordMatcherBean with the right attributes is returned.
+     * An empty list is used as the list of fields to perform the match.
+     */
+    @Test
+    public void testGetOperatorBean00() {
+        List<String> matchingFields = Collections.emptyList();
+        String keywords = "keyword";
+        String matchingType = KeywordMatchingType.CONJUNCTION_INDEXBASED.toString();
+        KeywordExtractPredicate keywordExtractPredicate = new KeywordExtractPredicate(matchingFields, keywords, matchingType);
+        
+        OperatorBean computedProjectionBean = keywordExtractPredicate.getOperatorBean("xxx");
```

Comment:
Since `getOperatorBean()` is NOT a getter, maybe it's better to rename it to "generateOperatorBean()" in a separate PR?